File: solutions/javascript/resistor-color-duo/1/resistor-color-duo.js

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];

export const decodedValue = (colors) => {
  return colors.slice(0,2).map((c) => COLORS.indexOf(c)).reduce((a,b) => a*10 + b, 0);
};

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
COLOR_CODES = {'Black': 0, 
            'Brown': 1, 
            'Red': 2, 
            'Orange': 3, 
            'Yellow': 4, 
            'Green': 5, 
            'Blue': 6, 
            'Violet': 7, 
            'Grey': 8, 
            'White': 9}

CODE_COLORS = dict((code,color) for color, code in COLOR_CODES.items())

def value(colors):
    return COLOR_CODES[colors[1].title()]+ 10 * COLOR_CODES[colors[0].title()]
